src/rewriter.fs

```diff
@@ -352,31 +352,57 @@ let squeezeBlockWithComma = function
         else stmt
     | stmt -> stmt
 
+let private simplifyBlock stmts = 
+    let b = stmts
+    // Avoid some optimizations when there are preprocessor directives.
+    let hasPreprocessor = Seq.exists (function Verbatim _ -> true | _ -> false) b
+
+    // Remove dead code after return/break/...
+    let endOfCode = Seq.tryFindIndex (function Jump _ -> true | _ -> false) b
+    let b = match endOfCode with
+            | Some x when not hasPreprocessor -> List.truncate (x+1) b
+            | _ -> b
+
+    // Multipurpose pass. Ideally, should be simplified and repeated until no more changes.
+    let hasNoDecl = List.forall (function Decl _ -> false | _ -> true)
+    let rec endsWithReturn = function
+        | Jump(JumpKeyword.Return, _) -> true
+        | Block stmts -> stmts |> List.last |> endsWithReturn
+        | _ -> false
+    let b = b |> List.collect (function
+        // Inline inner empty blocks without variable
+        | Block b when hasNoDecl b -> b
+        // if (c) return a(); else b();  ->  if (c) return a(); b();
+        | If (cond, bodyT, Some bodyF) when endsWithReturn bodyT ->
+            let bodyF = match bodyF with
+                        | Block b when hasNoDecl b -> b // inline inner empty blocks without variable
+                        | Decl _ as d -> [Block [d]] // a decl must stay isolated in a block, for the same reason
+                        | s -> [s]
+            List.Cons (If (cond, bodyT, None), bodyF)
+        // Remove "empty" declarations (of things that were inlined). This is not optional.
+        | Decl (_, []) -> []
+        | e -> [e])
+
+    // if(a)return b;return c;  ->  return a?b:c;
+    let rec replaceIfReturnsByReturnTernary = function
+        | If (cond, Jump(JumpKeyword.Return, Some retT), None) :: Jump(JumpKeyword.Return, Some retF) :: c ->
+            [Jump(JumpKeyword.Return, Some (FunCall(Op "?:", [cond; retT; retF])))]
```

Comment:
Ah, yes, looks like this isn't already working, because we are so conservative on when to use commas, because of compressed size again I suppose.

Reply:
I mean, if we were using squeezeBlockWithComma everywhere, rather than only when it saves some braces, it would already work.

Reply:
You'd need to run `squeezeBlockWithComma` on a subset of the block. You can do it, but it will make the minified output even less readable (unless you also update the Printer).